httpx/dispatch/http2.py

```diff
@@ -173,6 +185,13 @@ def initiate_connection(self) -> None:
                 logger.debug(
                     f"receive_event stream_id={event_stream_id} event={event!r}"
                 )
+                if isinstance(event, h2.events.WindowUpdated):
+                    if event_stream_id == 0:
+                        for window_update_event in self.window_update_received.values():
+                            window_update_event.set()
+                    else:
+                        self.window_update_received[event_stream_id].set()
```

Comment:
I think that `KeyError` can be raised here, if HTTP/2 on the server will not work correctly. Perhaps we should check that `event_stream_id` value is in the `window_update_received` dictionary.

Reply:
Handled this error in new commit.